### PR TITLE
Add clear button

### PR DESCRIPTION
**Pull request by hollow-owl**
Add a clear button that clears the board but doesn't reset the time, so you don' have to complete the Sudoku to restart it.

### CODE REVIEW

File: src/components/pages/Game/index.tsx [hunk 1]
```diff
@@ -235,6 +249,21 @@ class Game extends React.Component<PropsFromRedux> {
...
+    const clear = () => {
+      console.log("clearing");
+      this.props.clearGame(
+        this.props.game.sudokuId,
+        this.props.game.sudokuIndex,
+        this.props.game.difficulty,
+        this.props.game.timesSolved,
+        this.props.game.secondsPlayed,
+        this.props.game.previousTimes,
+      );
```
Comment (TN1ck) [inline src/components/pages/Game/index.tsx:261]: I'm wondering why you send these parameters along to clear the game, currently cleargame resets the following:

* activeCellCoordinates
* notesMode
* showCircleMenu
* showHints: false
* showWrongEntries
* showMenu
* showNotes
* state (sets it to paused)
* won (sets it to false)
(All of these are actually UI settings, which I should in generally put somewhere else, but that's another topic)

The others are provided in clearGame as parameters and are the same as the ones in the store.
As I see it the actually logic are the next lines, where you set the sudoku.

Or am I missing something?

File: src/components/pages/Game/index.tsx [hunk 2]
```diff
@@ -235,6 +249,21 @@ class Game extends React.Component<PropsFromRedux> {
       pauseGame();
       chooseGame();
     };
+    const clear = () => {
+      console.log("clearing");
```
Comment (TN1ck) [inline src/components/pages/Game/index.tsx:253]: remove this.

**Review by TN1ck — CHANGES_REQUESTED** [review]
Thanks so much for the PR!

I have some question about the need of clearGame and there is an obsolete console.log.

**Comment by hollow-owl** [comment]
I copied from restartGame, but I don't really know how it works. I removed everything but
```js
      const sudoku = SUDOKUS[this.props.game.difficulty][this.props.game.sudokuIndex];
      this.props.setSudoku(sudoku.sudoku, sudoku.solution);
      this.props.continueGame();
```
which seems to work.

**Comment by hollow-owl** [comment]
Usually there's a restart/clear function. Sometimes it restarts the timer sometimes it doesn't. I think an "Are you sure?" would be helpful

**Pull request merged**